lib/requester/core.js

```diff
@@ -229,7 +229,9 @@ var dns = require('dns'),
             // returning error synchronously causes uncaught error because listeners are not attached to error events
             // on socket yet
             return setImmediate(function () {
-                callback(null, resolvedAddr, resolvedFamily);
+                // When options.all is set, the callback expects array of addresses instead of a single address
+                return options.all ? callback(null, [{ address: resolvedAddr, family: resolvedFamily }]) :
```

Comment:
where is this documented ? reference for options.all will help

Reply:
Ref: for options.all - https://www.geeksforgeeks.org/node-js-dns-lookup-method/
Also mentioned the same in RCA.